### PR TITLE
Add Branding Preference Management core implementation

### PR DESCRIPTION
**Pull request by UdeshAthukorala**

### CODE REVIEW

File: components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+                            org.wso2.carbon.identity.configuration.mgt.core.*; version="${carbon.identity.package.import.version.range}",
+
+                            org.json; version="${json.wso2.version.range}",
+                        </Import-Package>
+                    </instructions>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
+</project>
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml:125]: Add new line.

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/util/BrandingPreferenceMgtUtils.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    private static final Log log = LogFactory.getLog(BrandingPreferenceMgtUtils.class);
+
+    /**
+     * Generate Branding Preference input stream.
+     *
+     * @param preferencesJSON JSON string of preferences.
+     * @return Input stream of the preferences JSON string.
+     * @throws UnsupportedEncodingException Unsupported Encoding Exception.
+     */
+    public static InputStream generatePreferenceInputStream(String preferencesJSON) throws
```
Comment (madurangasiriwardena) [inline components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/util/BrandingPreferenceMgtUtils.java:49]: Do we really need this method?

**Review by madurangasiriwardena — APPROVED** [review]

File: components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml (hunk 1 above)
Comment (UdeshAthukorala) [reply components/org.wso2.carbon.identity.branding.preference.management.core/pom.xml]: Address with https://github.com/wso2-extensions/identity-branding-preference-management/pull/1/commits/6eb4bd7ef9000f51f02453170ce89aa75ec12ab7

File: components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/util/BrandingPreferenceMgtUtils.java (hunk 2 above)
Comment (UdeshAthukorala) [reply components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/util/BrandingPreferenceMgtUtils.java]: Removed. Addressed with https://github.com/wso2-extensions/identity-branding-preference-management/pull/1/commits/6eb4bd7ef9000f51f02453170ce89aa75ec12ab7

**Pull request merged**